protocol/src/main/java/org/geysermc/mcprotocollib/network/packet/Packet.java

```diff
@@ -9,13 +9,14 @@
 public interface Packet {
 
     /**
-     * Gets whether the packet has handling priority.
-     * If the result is true, the packet will be handled immediately after being
-     * decoded.
+     * Gets whether the packet should run on an async game thread rather than blocking the network (Netty) thread.
+     * Packets that qualify for this are usually packets with an ensureRunningOnSameThread call at the top of their packet listener method in the Minecraft code.
+     * Packets which need extra attention because they aren't "fully" handled async are marked using.
+     * // GAME THREAD DETAIL comments in the MCProtocolLib code.
      *
-     * @return Whether the packet has priority.
+     * @return Whether the packet be handled async from the Netty thread.
      */
-    default boolean isPriority() {
+    default boolean shouldRunOnGameThread() {
```

Comment:
This is so it is easier to read together with mojang. Because when you use netty all things run on the netty thread by default. Minecraft explicitly marks every packet as ticked. The reaspn is that you could accidentally implicitly tick a packet that shouldn't be ticked.

Reply:
Ticking that packet that shouldn't be ticked is more dangerous than a non-ticked packet. Especially when it comes to protocols, order and state switching.

Reply:
In that case, shouldn't the method have no default body? Just to ensure that it's not "forgotten" but set on purpose?

Reply:
Yeah I just didn't wanna edit more classes ig lol. Would you like me to explicitly set it for each packet?

Reply:
Resolved in private chat, we're keeping this impl